main/src/test/scala/ParseKey.scala

```diff
@@ -8,166 +8,114 @@
 package sbt
 
 import Def.{ displayFull, displayMasked, ScopedKey }
-import sbt.internal.{ TestBuild, Resolve }
-import TestBuild._
-import sbt.internal.util.complete._
+import sbt.internal.{ TestBuild, Resolve }, TestBuild._
+import sbt.internal.util.complete.Parser
 
-import org.scalacheck._
-import Gen._
-import Prop._
-import Arbitrary.arbBool
+import org.scalacheck._, Arbitrary.arbitrary, Gen._, Prop._
 
 /**
  * Tests that the scoped key parser in Act can correctly parse a ScopedKey converted by Def.show*Key.
  * This includes properly resolving omitted components.
  */
 object ParseKey extends Properties("Key parser test") {
-  final val MaxKeys = 5
-  final val MaxScopedKeys = 100
-
-  implicit val gstructure = genStructure
-
-  property("An explicitly specified axis is always parsed to that explicit value") =
-    forAllNoShrink(structureDefinedKey) { (skm: StructureKeyMask) =>
-      import skm.{ structure, key, mask => mask0 }
-
+  property("An explicitly specified axis is always parsed to that explicit value") = forAll {
```

Comment:
Is there a reason why this was changed to `forAll` from `forAllNoShrink `?

Reply:
I think `forAllNoShrink` is a debug tool?

do you know why `forAllNoShrink` was committed?

I'm hoping that with this we'll eventually be able to solve https://github.com/sbt/sbt/issues/3893.

Reply:
Not exactly sure why the original opted for no shrink.
I know what causes #3893 by the way. It's the ambiguous task and subproject name. In the first case, you have subproject `q` and task `q`. We just need to reject that as not legal in sbt.